runtime/Cpp/runtime/src/atn/PredictionContext.cpp

```diff
@@ -252,130 +221,100 @@ Ref<PredictionContext> PredictionContext::mergeArrays(const Ref<ArrayPredictionC
   const Ref<ArrayPredictionContext> &b, bool rootIsWildcard, PredictionContextMergeCache *mergeCache) {
 
   if (mergeCache != nullptr) {
-    auto existing = mergeCache->get(a, b);
-    if (existing) {
-      return existing;
-    }
-    existing = mergeCache->get(b, a);
-    if (existing) {
+    if (const Ref<PredictionContext>& existing = mergeCache->get(a, b)) {
```

Comment:
You are ignoring the (b, a) combination here. Doesn't that significantly change the merge algorithm?

Reply:
I have added some comments to make this more clear. Lookup of the merge cache is now symmetric, 

`mergeCache->get(a, b) == mergeCache->get(b, a)`